dcrec/secp256k1/curve_test.go

```diff
@@ -56,6 +101,241 @@ func jacobianPointFromHex(x, y, z string) JacobianPoint {
+		x1:   "34f9460f0e4f08393d192b3c5133a6ba099aa0ad9fd54ebccfacdfa239ff49c6",
+		y1:   "0b71ea9bd730fd8923f6d25a7a91e7dd7728a960686cb5a901bb419e0f2ca232",
+		z1:   "1",
+		x2:   "34f9460f0e4f08393d192b3c5133a6ba099aa0ad9fd54ebccfacdfa239ff49c6",
+		y2:   "f48e156428cf0276dc092da5856e182288d7569f97934a56fe44be60f0d359fd",
+		z2:   "1",
+		want: false,
+	}, {
+		// Points that are opposites with z1=z2=2.
+		name: "P(x, y, 2) != P(x, -y, 2)",
+		x1:   "d3e5183c393c20e4f464acf144ce9ae8266a82b67f553af33eb37e88e7fd2718",
+		y1:   "5b8f54deb987ec491fb692d3d48f3eebb9454b034365ad480dda0cf079651190",
+		z1:   "2",
+		x2:   "d3e5183c393c20e4f464acf144ce9ae8266a82b67f553af33eb37e88e7fd2718",
+		y2:   "a470ab21467813b6e0496d2c2b70c11446bab4fcbc9a52b7f225f30e869aea9f",
+		z2:   "2",
+		want: false,
+	}, {
+		// Points with same x, opposite y, and different z values with z2=1.
+		name: "P(x, y, 2) != P(x, -y, 1)",
+		x1:   "d3e5183c393c20e4f464acf144ce9ae8266a82b67f553af33eb37e88e7fd2718",
+		y1:   "5b8f54deb987ec491fb692d3d48f3eebb9454b034365ad480dda0cf079651190",
+		z1:   "2",
+		x2:   "34f9460f0e4f08393d192b3c5133a6ba099aa0ad9fd54ebccfacdfa239ff49c6",
+		y2:   "f48e156428cf0276dc092da5856e182288d7569f97934a56fe44be60f0d359fd",
+		z2:   "1",
+		want: false,
+	}, {
+		// Points with same x, opposite y, and different z values with z!=1.
+		name: "P(x, y, 2) + P(x, -y, 3) = ∞",
+		x1:   "d3e5183c393c20e4f464acf144ce9ae8266a82b67f553af33eb37e88e7fd2718",
+		y1:   "5b8f54deb987ec491fb692d3d48f3eebb9454b034365ad480dda0cf079651190",
+		z1:   "2",
+		x2:   "dcc3768780c74a0325e2851edad0dc8a566fa61a9e7fc4a34d13dcb509f99bc7",
+		y2:   "cafc41904dd5428934f7d075129c8ba46eb622d4fc88d72cd1401452664add18",
+		z2:   "3",
+		want: false,
+	}, {
+		// Points with all different values.
+		name: "P(x1, y1, 2) + P(x2, y2, 1)",
+		x1:   "d3e5183c393c20e4f464acf144ce9ae8266a82b67f553af33eb37e88e7fd2718",
+		y1:   "5b8f54deb987ec491fb692d3d48f3eebb9454b034365ad480dda0cf079651190",
+		z1:   "2",
+		x2:   "d74bf844b0862475103d96a611cf2d898447e288d34b360bc885cb8ce7c00575",
+		y2:   "131c670d414c4546b88ac3ff664611b1c38ceb1c21d76369d7a7a0969d61d97d",
+		z2:   "1",
+		want: false,
+	}}
+
+	for _, test := range tests {
+		// Convert hex to Jacobian points.
+		p1 := jacobianPointFromHex(test.x1, test.y1, test.z1)
+		p2 := jacobianPointFromHex(test.x2, test.y2, test.z2)
+
+		// Ensure the test data is using points that are actually on the curve
+		// (or the point at infinity).
+		if !isValidJacobianPoint(&p1) {
+			t.Errorf("%s: first point is not on the curve", test.name)
+			continue
+		}
+		if !isValidJacobianPoint(&p2) {
+			t.Errorf("%s: second point is not on the curve", test.name)
+			continue
+		}
+
+		// Convert the points to affine and ensure they have the expected
+		// equivalency as well.
+		got := isSameAffinePoint(&p1, &p2)
+		if got != test.want {
+			t.Errorf("%s: mismatched expected test equivalency -- got %v, "+
+				"want %v", test.name, got, test.want)
+			continue
```

Comment:
we're obviously not hitting these but this continue (and the one below) aren't needed, and it might be "interesting" to see a failed test where only one them didn't match expected (instead of this one causing the public method to not be checked)

Reply:
Updated.

Reply:
It doesn't look like anything changed.

Reply:
Thanks.  Forgot to commit it.  Now it's done.